docs/source/best-practices.rst

```diff
@@ -0,0 +1,214 @@
+Best Practices
+==============
+
+It is easy to get started with Dask arrays, but using them *well* requires
+some experience.  This page contains suggestions for best practices, and
+includes solutions to common problems.
+
+This document specifically focuses on best practices that are shared among all
+of the Dask APIs.  Readers may first want to investigate one of the
+API-specific Best Practices documents first.
+
+-  :doc:`Arrays <array-best-practices>`
+-  :doc:`DataFrames <dataframe-best-practices>`
+-  :doc:`Delayed <delayed-best-practices>`
+
+
+Start Small
+-----------
+
+Parallelism brings extra complexity and overhead.
+Sometimes it's necessary for larger problems, but often it's not.
+Before adding a parallel computing system like Dask to your workload you may
+want to first try some alternatives:
+
+-   **Use better algorithms or data structures**:  NumPy, Pandas, Scikit-Learn
+    may have faster functions for what you're trying to do.  It may be worth
+    consulting with an expert or reading through their docs again to find a
+    better pre-built algorithm.
+
+-   **Better file formats**:  Efficient binary formats that support random
```

Comment:
I think it would be helpful to give some examples of some suggested file format types (later in this document Parquet, ORC, Zarr, HDF5, GeoTIFF, etc are mentioned). That makes it easier for someone new to dask to google and work out if it's appropriate for their situation or not. That's better than if people know there might be a problem but not knowing (a) if it's a problem in their particular case, or (b) how to address it.

Reply:
Thanks a ton for putting this all down on paper, btw! 
It's *really* useful to have and exactly the kind of thing I want to know about as a user.

Reply:
Thanks for the feedback @GenevieveBuckley !  I've cleaned up the language a bit below and added a link from here to there.  I didn't want to go in too much detail in this section (which talks about many things shallowly) but I hope that the link will get the interested reader to explore further.